File: r_csv/scrape_cars.R
# Load necessary libraries
library(rvest)
library(dplyr)

# Wikipedia page URL
url <- "https://en.wikipedia.org/wiki/Comma-separated_values"

# Read the HTML page
page <- read_html(url)

# Find the first table under "CSV example: cars"
cars_table <- page %>%
  html_node("table.wikitable") %>%
  html_table(fill = TRUE)

# Save to CSV locally
write.csv(cars_table, "cars.csv", row.names = FALSE)

print("CSV saved successfully!")

File: python_csv/scrape_cars.py
import requests
from bs4 import BeautifulSoup
import pandas as pd

# Wikipedia page URL
url = "https://en.wikipedia.org/wiki/Comma-separated_values"

# Get HTML content
response = requests.get(url)
soup = BeautifulSoup(response.text, "html.parser")

# Find the first table with class 'wikitable'
table = soup.find("table", {"class": "wikitable"})

# Parse table
rows = table.find_all("tr")
data = []

for row in rows:
    cols = row.find_all(["td", "th"])
    cols = [ele.text.strip() for ele in cols]
    data.append(cols)

# Convert to DataFrame
df = pd.DataFrame(data[1:], columns=data[0])

# Save CSV locally
df.to_csv("cars.csv", index=False)

print("CSV saved successfully!")
